src/app/src/test/kotlin/org/imperial/mrc/hint/integration/IntegrationTests.kt

```diff
@@ -1,13 +1,97 @@
 package org.imperial.mrc.hint.integration
 
+import org.assertj.core.api.Assertions
+import org.imperial.mrc.hint.helpers.AuthInterceptor
+import org.imperial.mrc.hint.helpers.JSONValidator
 import org.imperial.mrc.hint.helpers.tmpUploadDirectory
 import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.TestInfo
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.test.context.ActiveProfiles
 import java.io.File
 
+@ActiveProfiles(profiles = ["test"])
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 abstract class IntegrationTests {
 
     @AfterEach
     fun tearDown() {
         File(tmpUploadDirectory).deleteRecursively()
     }
-}
\ No newline at end of file
+}
+
+abstract class SecureIntegrationTests : IntegrationTests() {
+
+    @Autowired
+    lateinit var testRestTemplate: TestRestTemplate
+
+    @BeforeEach
+    fun beforeEach(info: TestInfo) {
+        val isAuthorized = info.displayName.contains("TRUE")
```

Comment:
This is the only way to get at a test's parameter values here - they are contained in the display name.